addons/web/static/src/js/views/form_relational_widgets.js

```diff
@@ -471,10 +471,22 @@ var AbstractManyField = common.AbstractField.extend({
         if (_.find(ids, function(id) { return typeof(id) === "string"; } )) {
             throw new Error("set_value of '"+this.name+"' must receive an list of ids without virtual ids.", ids);
         }
-        if (_.find(ids, function(id) { return typeof(id) !== "number"; } )) {
-            this.dataset.reset_ids([]);
+        if (_.find(ids, function(id) { return !(id instanceof Number); } )) {
+            this.dataset.reset_ids([], true);
+        }
+        if (_.find(ids, function(id) { return id instanceof Array; } )) {
             return this.send_commands(ids);
         }
+        if (_.find(ids, function(id) { return id instanceof Object; } )) {
+            return this.send_commands(_.map(ids, function (obj) {
+                if (obj.id) {
+                    return COMMANDS.update(obj.id, obj);
+                } else {
+                    return COMMANDS.create(obj);
+                }
+            }));
+        }
```

Comment:
Be careful: arrays are instances of `Object`! So if you invoke `set_value` with commands, you will execute both `if` statements, and the second one will probably crash.

I suggest to replace both `if` statements above by something like:

    return this.send_commands(_.map(ids, function (command) {
        if (command instanceof Array) {
            return command;
        } else if (command.id) {
            return COMMANDS.update(command.id, command);
        } else {
            return COMMANDS.create(command);
        }
    }));